src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -5301,6 +5301,222 @@ struct npc_silverpine_generic_actor_fenris_keep : public ScriptedAI
+    {
+        if (Player* player = killer->ToPlayer())
+        {
+            if (player->GetQuestStatus(QUEST_EXCISING_THE_TAINT) == QUEST_STATUS_NONE)
+            {
+                if (Quest const* excisingTheTaint = sObjectMgr->GetQuestTemplate(QUEST_EXCISING_THE_TAINT))
+                    player->AddQuest(excisingTheTaint, nullptr);
+            }
+        }
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_SNIFFING:
+                    me->HandleEmoteCommand(EMOTE_ONESHOT_SNIFF);
+                    _events.Repeat(18s, 25s);
+                    break;
+
+                case EVENT_STRIKE_BOAR:
+                    if (Creature* boar = me->FindNearestCreature(NPC_DARKTUSK_BOAR, 5.0f))
+                        me->CastSpell(boar, SPELL_BLOODY_STRIKE, true);
+                    _events.Repeat(5s, 7s);
+                    break;
+
+                case EVENT_SHADOWSTEP:
+                    DoCastVictim(SPELL_SHADOWSTEP);
+                    _events.Repeat(14s, 15s);
+                    break;
+
+                case EVENT_SINISTER_STRIKE:
+                    DoCastVictim(SPELL_SINISTER_STRIKE);
+                    _events.Repeat(6s, 8s);
+                    break;
+
+                default:
+                    break;
+            }
+        }
+
+        if (!UpdateVictim())
+            return;
+
+        DoMeleeAttackIfReady();
+    }
+
+private:
+    EventMap _events;
+};
+
+Position const SmithersResetPos[2] =
+{
+    { 144.355f, 1524.87f, 114.94f, 4.702f },
+    { 144.111f, 1519.69f, 120.3f, 1.7278f }
+};
+
+enum CaretakerSmithers
+{
+    SPELL_FACE_RIP                              = 84440,
+    SPELL_THROW_LANTERN                         = 81764,
+    SPELL_FRENZY                                = 81173,
+
+    EVENT_SMITHERS_THROW_LANTERN                = 1,
+    EVENT_SMITHERS_FRENZY                       = 2,
+    EVENT_SMITHERS_RESET_POS                    = 3,
+
+    TALK_SMITHERS_AGROO                         = 0,
+    TALK_SMITHERS_LANTERN                       = 1,
+    TALK_SMITHERS_FRENZY                        = 2,
+
+    POINT_BEFORE_JUMPING                        = 1
+};
+
+// 45219 - Caretaker Smithers
+struct npc_silverpine_caretaker_smithers : public ScriptedAI
+{
+    npc_silverpine_caretaker_smithers(Creature* creature) : ScriptedAI(creature) { }
+
+    void Reset() override
+    {
+        _events.Reset();
+
+        if (me->GetPosition() != me->GetHomePosition())
+            _events.ScheduleEvent(EVENT_SMITHERS_RESET_POS, 50ms);
+    }
+
+    void MovementInform(uint32 type, uint32 id) override
+    {
+        if (type != POINT_MOTION_TYPE)
+            return;
+
+        if (id == POINT_BEFORE_JUMPING)
+            me->GetMotionMaster()->MoveJump(SmithersResetPos[1], 7.5f, 7.5f, EVENT_JUMP, true);
```

Comment:
Wasn't there an issue when starting movement within Movement inform or did that get fixed? @Shauren